README.pod
=head1 NAME

decency - Anti-Spam fighting framework

=head1 DESCRIPTION

B<Comming Soon>

decency ..

=over

=item * is a highly configurable postfix policy daemon, SMTP content filter.

=item * aims to reduce your spam significantly.

=item * tries to be as easy as possible for the admin.

=item * is designed to work in small, one mailserver setups as well as in complex distributed multi mailserver, multi filter server environments.

=item * does not cost you a dime because it is free open source software.

=item * has an interface to implement existing anti-spam solutions.

=back

=head1 DEVELOPMENT STATE

decency is close to beta, but not yet there. However, it has already passed months of production (yes, i can run it, but you shouldnt) and real time tests survived. Some modules have less and some have more. It's mostly a matter of not having the time to write the docu, cleanup unused stuff and so on.

=head1 INSTALL

See L<http://www.decency-antispam.org/tutorial/install>

=head1 DOCUMENTATION

See L<http://www.decency-antispam.org/>

=head1 REQUIREMENTS

=head2 PERL

At least you need perl 5.10

=head3 MODULES

    Archive::Tar    1.40
    Cache::File
    Crypt::OpenSSL::RSA
    Data::Pager
    DateTime
    DBD::SQLite
    DBI
    DBIx::Connector
    Digest::MD5
    Digest::SHA
    Email::Valid
    File::Path    2.07
    File::Temp
    IO::String
    IO::YAML    0.08
    Mail::Field::Received
    MIME::Lite
    MIME::Parser
    MIME::QuotedPrint
    Module::Build      0.2805
    Mouse
    MouseX::NativeTraits
    Net::DNS::Resolver
    Net::Domain::TLD
    Net::LMTP
    Net::Netmask
    Net::Server::PreFork
    Net::SMTP
    Proc::ProcessTable
    Regexp::Common
    Regexp::IPv6
    SQL::Abstract::Limit
    Storable
    Test::More
    Time::HiRes
    YAML

=head3 OPTIONAL

Depending on which modules you plan to use

    # using MongoDB as database
    MongoDB => 0.35,
    
    # using LDAP as database
    Net::LDAP
    
    # using ClamAV
    ClamAV::Client
    
    # using a cache
    Cache::FastMmap
    Cache::Memcache
    Cache::Memcache::XS
    
    # should not be installed via perl .. rather by distri installer
    #Mail::SpamAssassin
    
    # should be installed via distri
    #Geo::IP



Changes
0.2.0 2011-11-29 Ulrich Kautz <uk@fortrabbit.de>
    * Renamed ContentFilter:: to Detective:: and Policy:: to Doorman::
    * Migrated shema definition to separate model modules
    * Using SQL::Abstract::Limit instead of SQL::Abstract
    * Database abstraction can now limit and offset
    * New database methods: distinct, count
    * Require now MongoDB >= 0.35 (if using mongo)
    * pageset for database (with Data::Pager)
    * Began web interface for decency (not in this package)
    * Greylist db changed (databases are now: sender address + ip + recipient address, sender address + ip + recipient domain, sender domain + ip + recipient domain)
    * using session data instead of attributes directly in all Doorman modules
    * Server stats extended: collect stream log and final response states/period
    * Renamed the server.pl script to decency.pl
    * Added daemon support, change of user and group, run-check method, cleanup database method and kill method for the server script
    * Wrote maintenance cron-scripts
    * Switched from Moose to Mouse reducing memory footprint to about 60% and gain speed up startup and execution
    * Restructuring Roles / Modules. Extended MouseX with new inheritance behaviour of roles.
    * Renamed the M::D::Core::Child module to M::D::Core::Module
    * Renamed the session_data (Server, Module) handler to session
    * Keeping track of file handles (open, close, temp files, static files). Modules which die due to timeout or other reasons cannot leave open handles.
    * MIME changes are now more efficient (not every module making changes has to write the changes immediately)
    * Splitted M::D::CF::DKIM in DKIMVerify and DKIMSign for clearification
    * IPC removal problem fixed
    * Added exclusions directory in addition to config, file and database
    * Renamed "policy"- and "filters"-attributes in server YAML files to "modules"
    * Generalized load_modules method for Detective and Doorman
    * Added LDAP as database
    * Added custom (domain or address based) thresholds via CustomScoring module
    * Dropped LogParser. Maybe revive it in the future
    * Switchted to Net::Server::PreFork instead of POE


0.1.7 2010-07-19 Ulrich Kautz <uk@fortrabbit.de>
    * M::D::Policy::Greylist: Repsonse line does not include other modules opinions anymore ..
    * M::D::Policy::SPF: Reject message only, if it is a rejection
    * M::D::Policy: add_response_message handling updated
    * Added File::Path >= 2.07 to dependency list (tester report)
    * Simplified shared code for CF and Policy in handling session to modules
    * Simplified hook management
    * Added finish hooks to policy
    * Sempahore can now be provided to database
    * M::D::ContentFilter::Archive can now be enabled to archive mails which are recognized as spam
    * Skip all tests, if DBD::SQLite is not available or IPC::Semaphore cannot create a semaphore

0.1.6 2010-07-12 Ulrich Kautz <uk@fortrabbit.de>
    * Updated description to clearify the goal of this approach
    * Updated M::D::LogParser::Core::PostfixParser docs for others as a background to implement other MTAs
    * Removed Geo::IP from dependencies, test will be skipped if not installed.
    * Added Net::DNSBL::Client as dependency in tests
    * Implemented DSPAM via Net::LMTP (instead of command line client)
    * Converted M::D::ContentFilter::Core::* to Mouse::Role
    * Implemented SpamAsssassin via Mail::SpamAssassin::Client
    * Enabled train mode for CF without cmd_-methods (DSPAM, SpamAssassin)

0.1.5 2010-07-02 Ulrich Kautz <uk@fortrabbit.de>
    * Association module passes mails which pass SPF
    * Added policy module: Basic (re-implement postfix basic checks, such as reverse hostname and hostname fqdn checks.. but with weighting)
    * Updated maintenance for M::D::Core::Stats
    * Added uninstall and install methods to init script
    * Added Mail::SPF, Email::Valid and Regexp::Common to PREREQ_PM
    * Added new M::D::Policy::Basic module for weighted postfix basic checks

0.1.4 2010-06-30 Ulrich Kautz <uk@fortrabbit.de>
    * Added the capability of printing SQL CREATE syntax via M::D::Core::DatabaseCreate module (server.pl .. --print-sql)
    * Updated Moduel documentation with correct SQL syntax
    * Updated tables.sql with current SQL (debian package)
    * empty FROM is now allowed (bounce..)
    * Reduced module "version" from 0.77 to 0.74, cause this is debian stable.
    * Updated some content filter documentations
    * Exclusions added (elide module per recipient/sender domain/address)

0.1.3 2010-06-25 Ulrich Kautz <uk@fortrabbit.de>
    * Tests fixed (28, 31, 45 and 67): skip if certain modules are not there
    * M::D::LogParser docu updated
    * Cookbooks updated (added Hook explanation)

0.1.2 2010-06-23 Ulrich Kautz <uk@fortrabbit.de>
    * forgot one last version ..
    * update M::D::ContentFilter description
    * update M::D::Policy description
    * added M::D::ContentFilter::Cookbook module for introduction
    * added M::D::Policy::Cookbook module for introduction

0.1.1 2010-06-23 Ulrich Kautz <uk@fortrabbit.de>
    * Forgot some versions..

0.1.0 2010-06-21 Ulrich Kautz <uk@fortrabbit.de>
    * Packed for CPAN
    * Reset version to 0.1.0



Build.PL
#
# Module::Build build script for Mail::SPF
#
# (C) 2005-2008 Julian Mehnle <julian@mehnle.net>
# $Id: Build.PL 50 2008-08-17 21:28:15Z Julian Mehnle $
#
##############################################################################

use Module::Build 0.26;
use version;

my $build = Module::Build->new(
    module_name => 'Mail::Decency',
    dist_author => [
        'Ulrich Kautz <uk@fortrabbit.de>',
    ],
    license => 'perl',
    requires => {
        'Archive::Tar' => 1.40,
        'Cache::File' => 0,
        'Crypt::OpenSSL::RSA' => 0,
        'Data::Pager' => 0,
        'DateTime' => 0,
        'DBD::SQLite' => 0,
        'DBI' => 0,
        'DBIx::Connector' => 0,
        'Digest::MD5' => 0,
        'Digest::SHA' => 0,
        'Email::Valid' => 0,
        'File::Path' => 2.07,
        'File::Temp' => 0,
        'IO::String' => 0,
        'IO::YAML' => 0.08,
        'Mail::Field::Received' => 0,
        'MIME::Lite' => 0,
        'MIME::Parser' => 0,
        'MIME::QuotedPrint' => 0,
        'Module::Build' => 0.2805,
        'Mouse' => 0,
        'MouseX::NativeTraits' => 0,
        'Net::DNS::Resolver' => 0,
        'Net::Domain::TLD' => 0,
        'Net::LMTP' => 0,
        'Net::Netmask' => 0,
        'Net::Server::PreFork' => 0,
        'Net::SMTP' => 0,
        'Proc::ProcessTable' => 0,
        'Regexp::Common' => 0,
        'Regexp::IPv6' => 0,
        'SQL::Abstract::Limit' => 0,
        'Storable' => 0,
        'Test::More' => 0,
        'Time::HiRes' => 0,
        'YAML' => 0,
    },
    recommends => {
        NetAddr::IP => '4.007'
    },
    build_requires => {
        Cache::File => 0,
        DBD::SQLite => 0,
        DBI => 0,
        DBIx::Connector => 0,
        SQL::Abstract::Limit => 0,
        Module::Build => '0.2805',
        Test::More => 0,
        
        # optional, for modules which can be tested
        Mail::SPF => 0,
        Mail::DKIM => 0,
        Net::DNSBL::Client => 0
    },
    optional => {
        
        # using MongoDB as database
        MongoDB => 0.35,
        
        # using LDAP as database
        Net::LDAP => 0,
        
        # using ClamAV
        ClamAV::Client => 0,
        
        # using a cache
        Cache::FastMmap => 0,
        Cache::Memcache => 0,
        Cache::Memcache::XS => 0,
        
        # Milter
        Sendmail::PMilter => 0,
        
        # should not be installed via perl .. rather by distri installer
        #Mail::SpamAssassin => 0,
        
        # should be installed via distri
        #Geo::IP => 0
    },
    sign => 1
);

$build->add_build_element($_)
    foreach qw(bin);

$build->create_build_script();
